File: OCH_Tao/ABC/B/353.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const K = parseInt(input[0].split(" ")[1], 10);
const A = input[1].split(" ").map(x => parseInt(x, 10));
let tmp = 0;
let cnt = 0;
for (const i of A) {
  if (K - tmp < i) {
    cnt++;
    tmp = 0;
  }
  tmp += i;
}
console.log(cnt + 1);

File: OCH_Tao/ABC/B/353.py
N,K = map(int,input().split())
A = list(map(int,input().split()))
tmp = 0
cnt = 0
for i in A:
  if K-tmp<i:
    cnt+=1
    tmp = 0
  tmp+=i
else:
  cnt+=1
print(cnt)